microsoft-edge/progressive-web-apps-chromium/how-to/app-actions.md

```diff
@@ -0,0 +1,397 @@
+
+1. Upgrade to Edge Stable 137 version.  (Edge 137 stable will be available after May 29, 2025.)
+
+1. Package your PWA for [PWABuilder.com](https://www.pwabuilder.com); see [Package your PWA for the Store](./microsoft-store.md#package-your-pwa-for-the-store) in _Publish a PWA to the Microsoft Store_.
+
+1. In **All settings**, scroll down to **Actions – Enable Actions** at the bottom.
+
+1. Make sure your PWA's web app manifest file (such as [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json)) has `protocol_handlers` and `share_target` fields.  PWABuilder.com will run a check, and if either field is missing from the app's web app manifest, the **Enable Actions** checkbox will be greyed out.
+
+1. In the **Actions** section, click **Choose File**, and upload the action definition manifest file (such as `ActionsManifest.json`) that you prepared for your PWA.
+
+1. Click the **Download** button.
+
+1. Unzip your download archive.  Then you can [publish your PWA to the Microsoft Store](./microsoft-store.md) and install the app from the Store, or install it directly by running `install.ps1`.
+
+1. Launch your PWA by clicking the PWA icon.
+
+
+<!-- ====================================================================== -->
+## Package a store PWA in Edge Canary
+
+The Stable release of Edge 137 will be available on May 29, 2025; see [Package a store PWA in Edge Stable](#package-a-store-pwa-in-edge-stable), above.
+
+Edge Canary gives a preview of the functionalities that's not available in stable yet.
+
+If you want to try out App Actions for PWAs before May 29, 2025, you can package your PWA in Edge Canary, as follows:
+
+1. Install [Edge Canary](https://www.microsoft.com/edge/download/insider) and upgrade to the latest version.
+
+1. Get the **Package ID**, **Publisher ID**, and **Publisher display name** of your PWA; see [Publish a PWA to the Microsoft Store](./microsoft-store.md).
+
+1. Go to [PWABuilder.Windows.Chromium tester](https://pwabuilder-windows-docker.azurewebsites.net/).
+
+1. Enter your PWA configs in the input box, for the following fields:
+
+   * `name`: Your PWA's name.
+   * `packageId`: Not needed if you won't publish the Canary-bind app<!-- todo: is "Canary-bind app" meaningful? 3x --> to the store; can be an empty string.
+   * `url`: Your PWA's url.
+   * `publisher` > `displayName`: The publisher's name.  Not needed if you won't publish the Canary-bind app to the store; can be an empty string.
+   * `publisher` > `commonName`: Your Publisher ID.  Not needed if you won't publish the Canary-bind app to the store; can be an empty string.
+   * `startUrl`: Your PWA's `startUrl`.
+
+   Example of a JSON listing:<!-- todo: which file?  where is this snippet from?  need a lead-in, to introduce each code snippet and give context: --><!-- todo: is this the complete file/ JSON object? -->
```

Comment:
Where is this listing from?  Generated as you fill in the form?  Is this an example app actions definition manifest file such as ActionsManifest.json but specialized for Canary?
Above each code snippet, need a lead-in giving file name; type of file; context.
@novac42